src/telemetry/metrics/record.rs

```diff
@@ -12,13 +12,14 @@ use super::transport;
 #[derive(Clone, Debug)]
 pub struct Record {
     metrics: Arc<Mutex<Root>>,
+    transports: transport::Registry,
 }
 
 // ===== impl Record =====
 
 impl Record {
-    pub(super) fn new(metrics: &Arc<Mutex<Root>>) -> Self {
-        Self { metrics: metrics.clone() }
+    pub(super) fn new(metrics: &Arc<Mutex<Root>>, transports: transport::Registry) -> Self {
+        Self { metrics: metrics.clone(), transports }
```

Comment:
Code style nit: now that two fields are being initialized here, this probably ought to be:
```rust
    Self { 
        metrics: metrics.clone(), 
        transports,
    }
```

Reply:
Ah, I see that this [changes back again](https://github.com/linkerd/linkerd2-proxy/pull/76/files#diff-d59b7d0e0356005002b7a6173a16b0c2R20) in your next PR, not that big a deal then.